src/index.ts

```diff
@@ -458,56 +455,84 @@ export namespace OneSignal {
     }
 
     /** Add a callback that fires when the native push permission changes. */
-    export function addPermissionChangedHandler(
+    export function addPermissionObserver(
       observer: (event: { permission: boolean }) => void,
     ) {
       if (!isNativeModuleLoaded(RNOneSignal)) return;
 
       isValidCallback(observer);
-      RNOneSignal.addPermissionChangedHandler();
+      RNOneSignal.addPermissionObserver();
       eventManager.addEventHandler<{ permission: boolean }>(
         PERMISSION_CHANGED,
         observer,
       );
     }
 
     /** Remove permission observer that have been previously added. */
-    export function removePermissionChangedHandler() {
+    export function removePermissionObserver() {
       if (!isNativeModuleLoaded(RNOneSignal)) return;
 
-      RNOneSignal.removePermissionChangedHandler();
+      RNOneSignal.removePermissionObserver();
       eventManager.clearEventHandler(PERMISSION_CHANGED);
     }
 
-    /** Sets a handler that will run whenever a notification is opened by the user. */
-    export function setNotificationClickHandler(
-      handler: (openedEvent: OpenedEvent) => void,
-    ) {
+    /** iOS Only.
+     * Returns the enum for the native permission of the device. It will be one of:
+     * OSNotificationPermissionNotDetermined, 
+     * OSNotificationPermissionDenied, 
+     * OSNotificationPermissionAuthorized, 
+     * OSNotificationPermissionProvisional - only available in iOS 12, 
+     * OSNotificationPermissionEphemeral - only available in iOS 14
+     * */
+    export function permissionNative() {
       if (!isNativeModuleLoaded(RNOneSignal)) return;
 
-      isValidCallback(handler);
-
-      RNOneSignal.setNotificationClickHandler();
-      eventManager.setEventHandler<OpenedEvent>(NOTIFICATION_CLICKED, handler);
+      if (Platform.OS === 'ios') {
+        return RNOneSignal.permissionNative();
+      } else {
+        return notificationPermission ? OSNotificationPermission.Authorized : OSNotificationPermission.Denied;
+      }
     }
 
     /**
-     * Sets the handler to run before displaying a notification while the app is in focus. Use this handler to read notification
-     * data and change it or decide if the notification should show or not.
-     * Note: this runs after the Notification Service Extension which can be used to modify the notification before showing it.
-     */
-    export function setNotificationWillShowInForegroundHandler(
-      handler: (event: NotificationReceivedEvent) => void,
-    ) {
+     * Add listeners for notification click and/or lifecycle events. */
+    export function addEventListener<K extends NotificationEventName>(event: K, listener: (event: NotificationEventTypeMap[K]) => void): void {
```

Comment:
I think we may want to consider a refactor of the current EventManager methods now that multiples for each observer and listener type can be added. If multiple notification click listeners are added, do they all get triggered?

Reply:
Addressed in [Update EventManager to allow for multiple Notification and IAM listeners](https://github.com/OneSignal/react-native-onesignal/pull/1532/commits/c544316490d54890e927749723132ce2ee7423a0)